ci/build.py

```diff
@@ -37,11 +41,54 @@
 import platform
 from copy import deepcopy
 from itertools import chain
-from subprocess import call, check_call
+from subprocess import call, check_call, check_output
 from typing import *
 from util import *
+try:
+    pip_install('docker')
+    import docker
+    import docker.models
+    import docker.errors
+    import signal
+    import atexit
+except ImportError as e:
+    logging.error(e)
+    logging.critical("Py docker is required to run: https://pypi.org/project/docker/ it can be installed with 'pip3 install docker'")
+    sys.exit(1)
+
+
+class Cleanup:
+    def __init__(self):
+        self.containers = set()
+
+    def add_container(self, container: docker.models.containers.Container):
+        assert isinstance(container, docker.models.containers.Container)
+        self.containers.add(container)
+
+    def remove_container(self, container: docker.models.containers.Container):
+        assert isinstance(container, docker.models.containers.Container)
+        self.containers.remove(container)
+
+    def _cleanup_containers(self):
+        if self.containers:
+            logging.warning("Cleaning up containers")
+        docker_client = docker.from_env()
+        try:
+            stop_timeout = int(os.environ.get("DOCKER_STOP_TIMEOUT", 3))
+        except Exception as e:
+            stop_timeout = 3
+        for container in self.containers:
+            try:
+                container.stop(timeout=stop_timeout)
+                logging.warning("☠: %s", trim_container_id(container.id))
```

Comment:
Should we have a more meaningful warning text than ☠?

Reply:
Do you think we need a warning level here?

Reply:
you suggest info?

Reply:
Yes, might be more appropriate